mlflow/models/signature.py

```diff
@@ -238,3 +245,76 @@ def _infer_signature_from_type_hints(func, input_arg_index, input_example=None):
     if input_schema is None and output_schema is None:
         return None
     return ModelSignature(inputs=input_schema, outputs=output_schema)
+
+
+def set_signature(
+    model_uri: str,
+    signature: ModelSignature,
+):
+    """
+    Sets the model signature for specified model artifacts.
+
+    The process involves downloading the MLmodel file in the model artifacts (if it's non-local),
+    updating its model signature, and then overwriting the existing MLmodel file. Should the
+    artifact repository associated with the model artifacts disallow overwriting, this function will
+    fail. Furthermore, as model registry artifacts are read-only, model artifacts located in the
+    model registry and represented by ``models:/`` URI schemes are not compatible with this API.
```

Comment:
Can we recommend that users create a new model version in order to update the signature? Might be nice to include a link to some docs about how to do that (we may have to add those docs as part of this PR too, or feel free to inline as a Python code example)

Reply:
I'll add a new doc section in the Model Signatures doc in a follow up PR.